sycl/test-e2e/Graph/Update/update_with_indices_accessor.cpp

```diff
@@ -0,0 +1,68 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// Extra run to check for leaks in Level Zero using UR_L0_LEAKS_DEBUG
+// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+// Extra run to check for immediate-command-list in Level Zero
+// RUN: %if level_zero && linux %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+//
+// UNSUPPORTED: opencl, level_zero
+
+// Tests updating a graph node scalar argument using index-based explicit update
+
+#include "../graph_common.hpp"
+
+int main() {
+  queue Queue{};
+
+  const size_t N = 1024;
+
+  exp_ext::command_graph Graph{
+      Queue.get_context(),
+      Queue.get_device(),
+      {exp_ext::property::graph::assume_buffer_outlives_graph{}}};
```

Comment:
It looks like we only have one accessor test, but for me the most complicated part of this PR for me to review is the changes around accessor requirements. It would be nice if we add a couple more accessor tests to have confidence in these changes.

For example. if we have a graph with multiple nodes, some of which use a buffer accessor directly and others nodes which use a buffer accessor wrapped in an dynamic_parameter can we test the correct dependencies are observed, and update works as expected.

Reply:
> For example. if we have a graph with multiple nodes, some of which use a buffer accessor directly and others nodes which use a buffer accessor wrapped in an dynamic_parameter can we test the correct dependencies are observed, and update works as expected.

I feel like the only way you could actually test that scenario is by having those nodes have no dependencies on each other, because only updating some nodes to depend on a different buffer would not create new edges or modify existing ones, so the dependencies will be wrong, and it doesn't seem like we should be testing that? But also have only root nodes doesn't seem to test what you want here I think.

Reply:
I have added another accessor test for double updating since that was an issue with the previous approach.

Reply:
> I feel like the only way you could actually test that scenario is by having those nodes have no dependencies on each other, because only updating some nodes to depend on a different buffer would not create new edges or modify existing ones, so the dependencies will be wrong, and it doesn't seem like we should be testing that? But also having only root nodes doesn't seem to test what you want here I think.

That's a good point, I guess really I'm just looking some more complex tests. Ideally (1) another test which uses buffer accessors and has more than one node in the graph, and (2) a test where the buffer accessors used in the graph interact with buffer accessors outside of the graph when the executable graph is submitted. Something like:
```cpp
  auto Acc = BufA.get_access();
  exp_ext::dynamic_parameter InputParam(Graph, Acc);
  
  auto KernelNode = Graph.add([&](handler &cgh) {
    cgh.require(InputParam);
    cgh.set_arg(0, InputParam);
    cgh.single_task(...);
    });
  });
  
  auto ExecGraph = Graph.finalize(exp_ext::property::graph::updatable{});

  Queue.submit(/* kernel with BufA */);
  
  // executable graph should have dependency on previous queue submission
  Queue.ext_oneapi_graph(ExecGraph).wait();
  
  // Swap BufB to be the input
  InputParam.update(BufB.get_access());
  ExecGraph.update(KernelNode);
  
  Queue.submit(/* kernel with BufB */ );

  //  Should executable graph have dependency on previous queue submission now we're using buffer B accessor?
  Queue.ext_oneapi_graph(ExecGraph).wait();
```

Reply:
LGTM, thanks.